### PR TITLE
Add meta-data to SutState

### PR DESCRIPTION
**Pull request by roesslerj**
- [x] the code follows the [Clean Code](https://clean-code-developer.com/) guidelines.
- [x] the necessary tests are either created or updated.
- [x] all status checks (Travis CI, SonarCloud, etc.) pass.
- [x] your commit history is cleaned up.
- [ ] you updated the changelog.
- [ ] you updated necessary documentation within [retest/docs](https://github.com/retest/docs).

### CODE REVIEW

File: src/main/java/de/retest/recheck/ui/descriptors/SutState.java [hunk 1]
```diff
@@ -22,25 +25,42 @@
 	@XmlElement
 	private final List<RootElement> descriptors;
 
+	@XmlElement
+	private final Map<String, String> metaData;
```
Comment (diba1013) [inline src/main/java/de/retest/recheck/ui/descriptors/SutState.java:29]: [Metadata](https://en.wikipedia.org/wiki/Metadata) is one word only:
```suggestion
	private final Map<String, String> metadata;
```

**Review by diba1013 — CHANGES_REQUESTED** [review]
Please take into consideration [RET-1896](https://retest.atlassian.net/browse/RET-1896) and its releated tasks.

Things to note:

1. Should we use a higher abstraction level (e.g. separate class)

    1. Keep it simple: No.
    2. Give it some thought: Yes. Then we could for example enforce a naming schema with namespaces and such for key collisions. Secondly, we do not have these confusing unmodifiable Map stuff, since we control the methods in it.
        ```
        @ToString
        @EqualsAndHashCode
        public class Metadata {

            Map<String, String> metadata;

            public Metadata(Map<Key, String> metadata) {
                this.metadata = // convert and flat the metadata
            }

            String get( final Key key );
            String getOrDefault( final Key key, String defaultValue );
        }

        @ToString
        @EqualsAndHashCode
        class Key {

            @Getter
            String key;

            public Key(String namespace, String part, String... additionalParts) {
                this( /* concat those with 'namespace:part1.part2' */ )
            }
        }
        ```
        And then use those `Key` to access the value behind.
2. This should be compatible with the old Golden Masters. However:

    1. Should we just in case increase the persistence version for that?
    2. How does JAXB handle loading of the old Golden Masters. Is the `metadata = null`?

File: src/main/java/de/retest/recheck/ui/descriptors/SutState.java (hunk 1 above)
Comment (roesslerj) [reply src/main/java/de/retest/recheck/ui/descriptors/SutState.java:29]: Done

**Comment by roesslerj** [comment]
Forgot to adapt the metadata retrieval...

**Pull request closed without merging**